akka-http/src/main/scala/akka/http/scaladsl/server/directives/TimeoutDirectives.scala

```diff
@@ -16,6 +15,16 @@ import scala.concurrent.duration.Duration
  */
 trait TimeoutDirectives {
 
+  def extractRequestTimeout: Directive1[Duration] = Directive { inner ⇒ ctx ⇒
+    val timeout = ctx.request.header[`Timeout-Access`] match {
+      case Some(t) ⇒ t.timeoutAccess.getTimeout
+      case _ ⇒
+        ctx.log.warning("extractRequestTimeout was used in route however no request-timeout is set!")
+        Duration.Inf
```

Comment:
Anyone have input on how this error case should be handled?

Reply:
`Duration.Inf` is fine, no need to log a warning. Let's document it on the directive.

Reply:
Is it a common case that the timeout access header is missing? What situation should we tell users to expect it?

Reply:
Ah, good point. I think there's currently no way the header would be missing. Maybe we can just fail in that case because it should be seldom enough?

Reply:
I think failing or continuing with `Duration.Inf` would both be fine, maybe bump the log to error level.
What would failure look like here? Completing with an exception?

Reply:
Maybe, let's keep it for now. We might want to allow switching off request timeouts completely for performance reasons later on, so not requiring that this header is always available is maybe the best solution.